### DISCUSSION TITLE
after use router.removeRoute('myName') next use router.getRoutes()  'myName' route-record still exit

### DISCUSSION THREAD

**Discussion by guda-art** [vuejs/router#2386] [Help and Questions] [answered]
### Reproduction

no content

### Steps to reproduce the bug

no content

### Expected behavior

delete route-record 

### Actual behavior

no-delete route-record 

### Additional information

I want to know if this is the default behavior. If not, it may be a side effect of my code
thanks

**Comment by guda-art** [comment]
exit => exist sorry

**Comment by posva** [comment, marked as answer]
[It seems to work](https://play.vuejs.org/#eNqlVNtu2zAM/RXNG5AES6Q0LfrguUG7olg37FJ0w17mPTi2kri1JUGS0xSB/33UxZe0aV+GwLFEHpKHOrR2QZnkDN+pIAzyUnCp0Q6lkiaaXgiBarSUvESDTUUHMWsBkleaytaLiTN0EBPb+BIhsIuPWZt5CM8oZgjhStGhC3f7kldMDwdvIWwwCsaBDwd+kaalKCB+boDR+mh+TYuCm2JvIgJbaxb2BQulJWer+WUlJWXacUYi0eswIt6Hdjv0zjrwsiqKG3CiurZpiMsTsWTTJLy1LL/m7B5pfhYHJA7mnzis0TUvaUQ6/4sRyQIsbdiF2T2Li0hTMzLa7Of6ndMHRDzOuyPSHgyclztLrA8o+o2WXD5e50rDa+yNLm9f6UmrZqunadCWbkRtDF7ZRnbT0B6utbQjQLcWmnKmvCgSne1xGe5Me2tHMzzEfTgaG4iNViH6445o59RFAzKA5jhQYiB82JGvbVQfaPXYR7eMx4glJQWUA/nov/Bfw6TGzB+0BGIbx3zooeA27fGC4oKv/HTjFdUWpYYjAIBS/TM0461SmQuNFNWVQEXCVjAxWsG0GImdE5awefIdzOZNJvgKZs8GYk+C/6vTpjpYyA3BpEwE3CecQSkrZOwdUCFE1mJs3ZwZcxystRYqJKRi4n6FQQ/SIc5P8DGekgzE71kxVeVkIfmDgvUdZPfqxsE5gEhGN5rzQk0Skb9U4hnw/BSf4iNS5AsC2UnOMrq1uU1quBhqaFMrEHeZr540aUYoL6j8IXQO4u81m8A19fDF2rSsaEs0XdP0/oD9Tm0d5RtJobsN7TWnEwmT5NxXP7/TLaxbZ8mzqgD0K85bCoNZGY4O9rFiGdDu4Szbz1aznK1+qautpkw1TRmi9jQs3gp5+UrrHd1jfNI7RaUfC6pwqsw9BRfeGJnbzMUtuMyoDNFMbBGQzTP0djqdfjCuEtLlbLLgWvMyREdTsbV2kWQZkG0tUCVmkBbNUYLew2MT++iCLuFD7yPXs37lLn1HIE3THoEQTeE38xmC+h8Xs3WG).

FYI: always provide a repro when opening issues
